### PR TITLE
Modifies the paths for deletion according to linux containers

### LINKED ISSUE #2126: Push, delete then push again doesn't propagate the deletions on windows platform
[kind/bug]

## What versions of software are you using?
- Operating System: Windows10
- Output of `odo version`: master

## How did you run odo exactly?
```
odo push command tests when push command is executed
  should be able to create a file, push, delete, then push again propagating the deletions
  C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:57
Creating a new project: alhhkmhjsk
Running odo.exe with args [odo project create alhhkmhjsk -w -v4]
[odo]  -  Waiting for project to come up  ...
 V  Waiting for project to come up [646ms]
[odo]  V  Project 'alhhkmhjsk' is ready for use
[odo]  V  New project created and now using project : alhhkmhjsk
[odo] I0913 05:48:24.299026    2732 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
Created dir: C:\Users\User\AppData\Local\Temp\105353870
Running git.exe with args [git clone https://github.com/openshift/ruby-ex C:\Users\User\AppData\Local\Temp\105353870/ruby-ex]
[git] Cloning into 'C:\Users\User\AppData\Local\Temp\105353870/ruby-ex'...
Running odo.exe with args [odo component create ruby nodejs --project alhhkmhjsk --context C:\Users\User\AppData\Local\Temp\105353870/ruby-ex --app app]
 V  Validating component [448ms]
[odo] Please use `odo push` command to create the component with source deployed
Running odo.exe with args [odo push --context C:\Users\User\AppData\Local\Temp\105353870/ruby-ex]
[odo] Validation
 V  Checking component [714ms]
[odo]
[odo] Configuration changes
[odo]  V  Initializing component
 V  Creating component [2s]
[odo]
[odo] Pushing to component nodejs of type local
 V  Checking files for pushing [9ms]
 V  Waiting for component to start [28s]
 V  Syncing files to the component [5s]
 V  Building component [9s]
[odo]  V  Changes successfully pushed to component
Running oc.exe with args [oc get pods --namespace alhhkmhjsk --show-labels]
[oc] NAME                  READY   STATUS      RESTARTS   AGE   LABELS
[oc] nodejs-app-1-deploy   0/1     Completed   0          43s   openshift.io/deployer-pod-for.name=nodejs-app-1
[oc] nodejs-app-1-fmqn5    1/1     Running     0          41s   deployment=nodejs-app-1,deploymentconfig=nodejs-app
Running oc.exe with args [oc get dc nodejs-app --namespace alhhkmhjsk -o jsonpath='{range .spec.template.spec.containers[0].env[*]}{.name}:{.value}{"\n"}{end}']
[oc] 'ODO_S2I_SCRIPTS_URL:/usr/libexec/s2i
[oc] ODO_S2I_SCRIPTS_PROTOCOL:image://
[oc] ODO_S2I_SRC_BIN_PATH:/tmp
[oc] ODO_S2I_DEPLOYMENT_DIR:
[oc] ODO_S2I_WORKING_DIR:/opt/app-root/src
[oc] ODO_S2I_BUILDER_IMG:rhscl/ruby-25-rhel7
[oc] ODO_SRC_BACKUP_DIR:/opt/app-root/src-backup
[oc] 'Running oc.exe with args [oc exec nodejs-app-1-fmqn5 --namespace alhhkmhjsk -- ls -lai /tmp//src]
[oc] total 56
[oc] 125952914 drwxr-xr-x. 4 1000930000 root   134 Sep 13 12:49 .
[oc] 234945614 drwxrwxrwt. 1 root       root    31 Sep 13 12:49 ..
[oc] 125952915 -rw-r--r--. 1 1000930000 root     0 Sep 13 12:49 .dummy
[oc]  16930709 drwxr-xr-x. 2 1000930000 root    52 Sep 13 12:49 .odo
[oc] 125952916 -rw-r--r--. 1 1000930000 root    55 Sep 13 12:49 Gemfile
[oc] 125952917 -rw-r--r--. 1 1000930000 root   165 Sep 13 12:49 Gemfile.lock
[oc] 125952918 -rw-r--r--. 1 1000930000 root  3614 Sep 13 12:49 README.md
[oc] 125952919 -rw-r--r--. 1 1000930000 root 40524 Sep 13 12:49 config.ru
[oc] 125952920 -rw-r--r--. 1 1000930000 root    11 Sep 13 12:49 foobar.txt
[oc]  25553991 drwxr-xr-x. 2 1000930000 root     6 Sep 13 12:49 testdir
Deleting dir: C:\Users\User\AppData\Local\Temp\105353870\ruby-ex\foobar.txt
Deleting dir: C:\Users\User\AppData\Local\Temp\105353870\ruby-ex\testdir
Running odo.exe with args [odo push --context C:\Users\User\AppData\Local\Temp\105353870/ruby-ex -v4]
[odo] I0913 05:49:20.383620    1496 common_push.go:119] SourceLocation: ./
[odo] I0913 05:49:20.384783    1496 common_push.go:127] Source Path: C:\Users\User\AppData\Local\Temp\105353870\ruby-ex
[odo] Validation
[odo]  -  Checking component  ...
[odo] I0913 05:49:21.504456    1496 occlient.go:2639] Getting DeploymentConfig: nodejs-app
[odo] I0913 05:49:22.196840    1496 component.go:513] Checking source location: ./
 V  Checking component [692ms]
[odo]
[odo] Configuration changes
[odo]  -  Retrieving component data  ...
[odo] I0913 05:49:22.645960    1496 occlient.go:2639] Getting DeploymentConfig: nodejs-app
[odo] I0913 05:49:22.868955    1496 component.go:957] Source for component nodejs is file:///./ (local)
[odo] I0913 05:49:22.868955    1496 occlient.go:2639] Getting DeploymentConfig: nodejs-app
[odo] I0913 05:49:23.098245    1496 component.go:1082] Updating component nodejs, from local to ./ (local).
 V  Retrieving component data [901ms]
[odo]  -  Applying configuration  ...
[odo] I0913 05:49:23.544459    1496 occlient.go:762] Found exact image tag match for ruby:latest
 V  Applying configuration [907ms]
[odo] I0913 05:49:24.005373    1496 url.go:80] Listing routes with label selector: app.kubernetes.io/part-of=app,app.kubernetes.io/instance=nodejs
[odo] I0913 05:49:24.005373    1496 occlient.go:2508] Listing routes with label selector: app.kubernetes.io/part-of=app,app.kubernetes.io/instance=nodejs
[odo]
[odo] Pushing to component nodejs of type local
[odo] I0913 05:49:24.240692    1496 util.go:570] ignoring path C:/Users/User/AppData/Local/Temp/105353870/ruby-ex/.git because of glob rule C:/Users/User/AppData/Local/Temp/105353870/ruby-ex/.git
[odo] I0913 05:49:24.240692    1496 file_indexer.go:123] .odo or .git directory detected, skipping it
[odo] I0913 05:49:24.240692    1496 file_indexer.go:159] file deleted: testdir
[odo] I0913 05:49:24.240692    1496 file_indexer.go:159] file deleted: foobar.txt
[odo] I0913 05:49:24.248542    1496 common_push.go:191] List of files to be deleted: +[testdir foobar.txt]
[odo] I0913 05:49:24.248542    1496 common_push.go:214] Copying directory C:\Users\User\AppData\Local\Temp\105353870\ruby-ex to pod
[odo] I0913 05:49:24.248542    1496 component.go:635] PushLocal: componentName: nodejs, applicationName: app, path: C:\Users\User\AppData\Local\Temp\105353870\ruby-ex, files: [], delFiles: [testdir foobar.txt], isForcePush: false
 V  Checking file changes for pushing [12ms]
[odo] I0913 05:49:24.472148    1496 occlient.go:1761] Waiting for deploymentconfig=nodejs-app pod
[odo]  -  Waiting for component to start  ...
[odo] I0913 05:49:24.697580    1496 occlient.go:1785] Status of nodejs-app-1-fmqn5 pod is Running
[odo] I0913 05:49:24.698633    1496 occlient.go:1789] Pod nodejs-app-1-fmqn5 is Running
[odo] I0913 05:49:24.698633    1496 component.go:674] propogating deletion of files testdir foobar.txt to pod
[odo] I0913 05:49:24.698633    1496 occlient.go:3213] s2ipaths marked for deletion are [\tmp\src\testdir \tmp\src\foobar.txt \opt\app-root\src\testdir \opt\app-root\src\foobar.txt \opt\app-root\src-backup\src\testdir \opt\app-root\src-backup\src\foobar.txt]
 V  Waiting for component to start [225ms]
[odo]  -  Syncing files to the component  ...
 V  Syncing files to the component [2s]
[odo]  -  Building component  ...
[odo] + set -eo pipefail
[odo] + export ODO_UTILS_DIR=/opt/odo
[odo] + ODO_UTILS_DIR=/opt/odo
[odo] + export ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
[odo] + ODO_IMAGE_MAPPINGS_FILE=/opt/odo/language-scripts/image-mappings.json
[odo] ++ /opt/odo/bin/getlanguage
[odo] + IMAGE_LANG=
[odo] + '[' '!' -z '' ']'
[odo] + '[' '!' -z /opt/app-root/src ']'
[odo] + '[' /tmp '!=' /opt/app-root/src ']'
[odo] + '[' -n /opt/app-root/src-backup ']'
[odo] + '[' '!' -d /opt/app-root/src-backup ']'
[odo] + rsync -rlO /tmp/src/. /opt/app-root/src-backup/src/
[odo] + b_IFS='
[odo] '
[odo] + b_OFS=
[odo] + OIFS='
[odo] '
[odo] + IFS='
[odo] '
[odo] ++ ls -A /tmp/src/
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/.dummy
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/.odo
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/Gemfile
[odo] ---> Installing application source ...
[odo] ---> Building your Ruby application from source ...
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/Gemfile.lock
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/README.md
[odo] ---> Running 'bundle install --retry 2 --deployment --without development:test' ...
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/config.ru
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/foobar.txt
[odo] + for file in '`ls -A ${ODO_S2I_SRC_BIN_PATH}/src/`'
[odo] + rm -fr /opt/app-root/src/testdir
[odo] + OIFS=
[odo] + IFS='
[odo] '
[odo] + '[' -n rhscl/ruby-25-rhel7 ']'
[odo] + '[' rhscl/ruby-25-rhel7 == redhat-openjdk-18/openjdk18-openshift ']'
[odo] + '[' -f /tmp/src/.s2i/bin/assemble ']'
[odo] + '[' -n /usr/libexec/s2i ']'
[odo] + rm -rf /opt/app-root/src/.git
[odo] + /usr/libexec/s2i/assemble
[odo] Warning: the running version of Bundler (1.16.1) is older than the version that created the lockfile (1.16.4). We suggest you upgrade to the latest version of Bundler by running `gem install bundler`.
[odo] Using bundler 1.16.1
[odo] Using puma 3.12.0
[odo] Using rack 2.0.6
[odo] Bundle complete! 2 Gemfile dependencies, 3 gems now installed.
[odo] Gems in the groups development and test were not installed.
[odo] Bundled gems are installed into `./bundle`
[odo] ---> Cleaning up unused ruby gems ...
[odo] Running `bundle clean --verbose` with bundler 1.16.1
[odo] Warning: the running version of Bundler (1.16.1) is older than the version that created the lockfile (1.16.4). We suggest you upgrade to the latest version of Bundler by running `gem install bundler`.
[odo] Frozen, using resolution from the lockfile
[odo] + '[' '!' -z /opt/app-root/src ']'
[odo] + '[' -n /opt/app-root/src-backup ']'
[odo] + '[' /tmp '!=' /opt/app-root/src ']'
[odo] + '[' '' '!=' /opt/app-root/src ']'
[odo] + rsync -rlO /opt/app-root/src-backup/src/. /tmp/src/
[odo] + /opt/odo/bin/supervisord ctl stop run
[odo] run: stopped
[odo] + /opt/odo/bin/supervisord ctl start run
[odo] run: started
 V  Building component [5s]
[odo]  V  Changes successfully pushed to component
[odo] I0913 05:49:31.254207    1496 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
Running oc.exe with args [oc exec nodejs-app-1-fmqn5 --namespace alhhkmhjsk -- ls -lai /tmp//src]
[oc] total 56
[oc] 125952914 drwxr-xr-x. 4 1000930000 root   134 Sep 13 12:49 .
[oc] 234945614 drwxrwxrwt. 1 root       root    31 Sep 13 12:49 ..
[oc] 125952915 -rw-r--r--. 1 1000930000 root     0 Sep 13 12:49 .dummy
[oc]  33637076 drwxr-xr-x. 2 1000930000 root    52 Sep 13 12:49 .odo
[oc] 125952916 -rw-r--r--. 1 1000930000 root    55 Sep 13 12:49 Gemfile
[oc] 125952917 -rw-r--r--. 1 1000930000 root   165 Sep 13 12:49 Gemfile.lock
[oc] 125952918 -rw-r--r--. 1 1000930000 root  3614 Sep 13 12:49 README.md
[oc] 125952919 -rw-r--r--. 1 1000930000 root 40524 Sep 13 12:49 config.ru
[oc] 125952920 -rw-r--r--. 1 1000930000 root    11 Sep 13 12:49 foobar.txt
[oc]  41975071 drwxr-xr-x. 2 1000930000 root     6 Sep 13 12:49 testdir
Deleting project: alhhkmhjsk
Running odo.exe with args [odo project delete alhhkmhjsk -f]
[odo] This project contains the following applications, which will be deleted
[odo] Application app
[odo] This application has following components that will be deleted
[odo] component named nodejs
[odo] No services / could not get services
 V  Deleting project alhhkmhjsk [6s]
[odo]  V  Deleted project : alhhkmhjsk
Deleting dir: C:\Users\User\AppData\Local\Temp\105353870
+ Failure [83.313 seconds]
odo push command tests
C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:15
  when push command is executed
  C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:40
    should be able to create a file, push, delete, then push again propagating the deletions [It]
    C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:57
    Expected
        <string>: total 56
        125952914 drwxr-xr-x. 4 1000930000 root   134 Sep 13 12:49 .
        234945614 drwxrwxrwt. 1 root       root    31 Sep 13 12:49 ..
        125952915 -rw-r--r--. 1 1000930000 root     0 Sep 13 12:49 .dummy
         33637076 drwxr-xr-x. 2 1000930000 root    52 Sep 13 12:49 .odo
        125952916 -rw-r--r--. 1 1000930000 root    55 Sep 13 12:49 Gemfile
        125952917 -rw-r--r--. 1 1000930000 root   165 Sep 13 12:49 Gemfile.lock
        125952918 -rw-r--r--. 1 1000930000 root  3614 Sep 13 12:49 README.md
        125952919 -rw-r--r--. 1 1000930000 root 40524 Sep 13 12:49 config.ru
        125952920 -rw-r--r--. 1 1000930000 root    11 Sep 13 12:49 foobar.txt
         41975071 drwxr-xr-x. 2 1000930000 root     6 Sep 13 12:49 testdir
    not to contain substring
        <string>: foobar.txt
    C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:91
```

## Actual behavior
Test fails - https://github.com/openshift/odo/blob/master/tests/integration/cmd_push_test.go#L57

## Expected behavior
test scenario should work - https://github.com/openshift/odo/blob/master/tests/integration/cmd_push_test.go#L57

### PR DESCRIPTION
**Pull request by mik-dass**
fixes #2126 

How to test on windows:
1. create a component
2. add a file, push the component
3. delete the file, push and verify if the file is deleted in the container

### CODE REVIEW

**Comment by mik-dass** [comment]
```
Verify you have provided correct credentials.
[odo]  ✗  challenger chose not to retry the request
```
/retest

**Comment by amitkrout** [comment]
> ```
> Verify you have provided correct credentials.
> [odo]  ✗  challenger chose not to retry the request
> ```
> 
> /retest

Seems an issue/kind of flake in login test file. Ginkgo flag ```-randomizeAllSpecs``` runs random spec in sequentially causing the failure because In login test the spec are dependent to each other. will fix it.

**Comment by mik-dass** [comment]
```
Verify you have provided correct credentials.
[odo]  ✗  challenger chose not to retry the request
```
/retest

File: pkg/occlient/occlient.go [hunk 1]
```diff
@@ -3168,7 +3168,9 @@ func (c *Client) PropagateDeletes(targetPodName string, delSrcRelPaths []string,
 	}
 	for _, s2iPath := range s2iPaths {
 		for _, delRelPath := range delSrcRelPaths {
-			rmPaths = append(rmPaths, filepath.Join(s2iPath, delRelPath))
+			// since the paths inside the container are linux oriented
+			// so we convert the paths accordingly
+			rmPaths = append(rmPaths, filepath.ToSlash(filepath.Join(s2iPath, delRelPath)))
```
Comment (cdrage) [inline pkg/occlient/occlient.go]: Got to have @kadel 's opinion on this, but this looks correct to me..

**Comment by amitkrout** [comment]
@mik-dass #2164 has been fixed. please rebase your pr

**Comment by amitkrout** [comment]
Push command test script works fine on win10+4.2 cluster.
```
[...]
+ [SLOW TEST:104.912 seconds]
odo push command tests
C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:15
  when running odo push with flag --show-log
  C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:301
    should be able to spam odo push without anything breaking
    C:/Users/User/go/src/github.com/openshift/odo/tests/integration/cmd_push_test.go:302
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Ran 9 of 100 Specs in 1381.533 seconds
SUCCESS! -- 9 Passed | 0 Failed | 0 Pending | 91 Skipped
PASS

Ginkgo ran 1 suite in 23m18.0623655s
Test Suite Passed
PS C:\Users\User\go\src\github.com\openshift\odo>
```

**Comment by mik-dass** [comment]
> @mik-dass #2164 has been fixed. please rebase your pr

Done

**Comment by amitkrout** [comment]
```
Setting current dir to: 
Deleting dir: /tmp/936357313
• Failure [70.316 seconds]
odo sub component command tests
/go/src/github.com/openshift/odo/tests/integration/cmd_cmp_sub_test.go:13
  Creating component with numeric named context
  /go/src/github.com/openshift/odo/tests/integration/component.go:579
    should create default named component in a directory with numeric name []
    /go/src/github.com/openshift/odo/tests/integration/component.go:596
    Unexpected error:
      
        <*os.PathError | 0xc00023c570>: {Op: "chdir", Path: "", Err: 0x2}
        chdir : no such file or directory
    occurred
    /go/src/github.com/openshift/odo/tests/helper/helper_filesystem.go:42
```
This flake has been fixed in #2185 
@mik-dass Can you please rebase it again

**Comment by mik-dass** [comment]
```
This flake has been fixed in #2185
@mik-dass Can you please rebase it again
```
@amitkrout done

**Comment by amitkrout** [comment]
```
[odo] Configuration changes
[odo]  ✓  Initializing component
[odo]  •  Creating component  ...
[odo]  ✓  Creating component [303ms]
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [95ms]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [1m]
[odo]  •  Deploying component testcmp  ...
[odo]  ✗  Deploying component testcmp [5m]
[odo]  ✗  Failed to create component with name testcmp. Please use `odo config view` to view settings used to create component. Error: Timed out waiting for annotation to update
[odo] github.com/openshift/odo/pkg/occlient.(*Client).WaitAndGetDC
[odo] 	/go/src/github.com/openshift/odo/pkg/occlient/occlient.go:1744
[odo] github.com/openshift/odo/pkg/component.Deploy
[odo] 	/go/src/github.com/openshift/odo/pkg/component/component.go:831
[odo] github.com/openshift/odo/pkg/component.CreateComponent
```
/test v4.2-integration

**Comment by kadel** [comment]
Paths in FileIndex on Windows are still saved in some weird format :-(

Why the double slashes? What if someone checks out this file on a Mac or Linux?

It might be better to normalize path into one common format (the POSIX way) before it is recorded to the FileIndex. Converting it to the os specific format only when it is being used.

```
{
    "kind": "FileIndex",
    "apiVersion": "v1",
    "Files": {
        ".gitignore": {
            "Size": 29,
            "LastModifiedDate": "2019-09-25T13:27:24.2690408+02:00"
        },
        "README.md": {
            "Size": 13155,
            "LastModifiedDate": "2019-09-25T13:27:24.2690408+02:00"
        },
        "helm": {
            "Size": 0,
            "LastModifiedDate": "2019-09-25T13:27:24.2690408+02:00"
        },
        "helm\\nodejs": {
            "Size": 0,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "helm\\nodejs\\Chart.yaml": {
            "Size": 355,
            "LastModifiedDate": "2019-09-25T13:27:24.2690408+02:00"
        },
        "helm\\nodejs\\templates": {
            "Size": 4096,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "helm\\nodejs\\templates\\NOTES.txt": {
            "Size": 231,
            "LastModifiedDate": "2019-09-25T13:27:24.2690408+02:00"
        },
        "helm\\nodejs\\templates\\buildconfig.yaml": {
            "Size": 1076,
            "LastModifiedDate": "2019-09-25T13:27:24.2690408+02:00"
        },
        "helm\\nodejs\\templates\\deploymentconfig.yaml": {
            "Size": 1257,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "helm\\nodejs\\templates\\imagestream.yaml": {
            "Size": 183,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "helm\\nodejs\\templates\\route.yaml": {
            "Size": 216,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "helm\\nodejs\\templates\\service.yaml": {
            "Size": 280,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "helm\\nodejs\\values.yaml": {
            "Size": 1178,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift": {
            "Size": 0,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift\\pipeline": {
            "Size": 0,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift\\pipeline\\Jenkinsfile": {
            "Size": 173,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift\\pipeline\\README.md": {
            "Size": 436,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift\\templates": {
            "Size": 4096,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift\\templates\\nodejs-mongodb-persistent.json": {
            "Size": 16954,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift\\templates\\nodejs-mongodb.json": {
            "Size": 16604,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "openshift\\templates\\nodejs.json": {
            "Size": 8682,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "package.json": {
            "Size": 851,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "server.js": {
            "Size": 3939,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "tests": {
            "Size": 0,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "tests\\app_test.js": {
            "Size": 742,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "views": {
            "Size": 0,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        },
        "views\\index.html": {
            "Size": 41247,
            "LastModifiedDate": "2019-09-25T13:27:24.2846634+02:00"
        }
    }
}
```

**Comment by kadel** [comment]
I'll open a new issue for the above comment, it can be addressed in a separate PR. We can merge this one as it fixes one of the problems.

**Comment by amitkrout** [comment]
> I'll open a new issue for the above comment, it can be addressed in a separate PR. We can merge this one as it fixes one of the problems.
> 
> /approve

@kadel We had a follow up issue #1990 for test validation of ```odo-file-index.json```. Followup issue never works if it is not fixed immediately or not fixed in the same pr.

**Comment by mik-dass** [comment]
```
level=fatal msg="failed to initialize the cluster: Multiple errors are preventing progress:\n* Cluster operator monitoring is still updating\n* Cluster operator openshift-apiserver has not yet reported success\n* Cluster operator openshift-samples is still updating\n* Could not update authentication \"cluster\" (11 of 349)\n* Could not update deployment \"openshift-machine-api/machine-api-operator\" (76 of 349)\n* Could not update oauthclient \"console\" (219 of 349)\n* Could not update operatorgroup \"openshift-operators/global-operators\" (259 of 349)\n* Could not update operatorsource \"openshift-marketplace/certified-operators\" (279 of 349)\n* Could not update secret \"openshift-authentication/v4-0-config-system-ocp-branding-template\" (105 of 349)\n* Could not update serviceca \"cluster\" (290 of 349)\n* Could not update servicemonitor \"openshift-authentication/oauth-openshift\" (321 of 349)\n* Could not update servicemonitor \"openshift-controller-manager/openshift-controller-manager\" (349 of 349): timed out waiting for the condition"
```
/retest

**Pull request merged**